.github/update_packages.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/**
+ * @file
+ * Script to update package versions in a YAML configuration file.
+ */
+
+require 'vendor/autoload.php';
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Fetches the latest version of a package from Packagist or Drupal.org.
+ *
+ * @param string $packageName
+ *   The name of the package.
+ *
+ * @return string|null
+ *   The latest version string, or NULL if not found.
+ */
+function get_latest_version($packageName) {
+  $client = new Client();
+  $url = "https://repo.packagist.org/p/{$packageName}.json";
+
+  try {
+    $response = $client->get($url);
+    $data = json_decode($response->getBody(), TRUE);
+
+    $versions = array_keys($data['packages'][$packageName]);
+    usort($versions, 'version_compare');
+
+    $latestVersion = end($versions);
+
+    // Extract major.minor version (e.g., "13.3.3" becomes "13.x")
+    $versionParts = explode('.', $latestVersion);
+    if (count($versionParts) > 1) {
+      return $versionParts[0] . '.x';
+    }
+
+    return NULL;
+  }
+  catch (RequestException $e) {
+    return get_latest_version_from_drupal_org($packageName);
+  }
+}
```

Comment:
This would be much more readable if you would extract the Packagist part this into a separate function just like you did with `get_latest_version_from_drupal_org()`. Consider how much clearer this would be, for example:

```php
function get_latest_version($packageName) {
  return get_latest_version_from_packagist($packageName)
    ?? get_latest_version_from_drupal_org($packageName);
}
```